src/public/src/scripts/react/components/administrator/showCompanies.tsx

```diff
@@ -2,10 +2,8 @@ import * as React from "react";
 import {Link} from "react-router";
 import Navbar from "../../navbar/navbarSuperAdmin";
 import store from "../../../stores/companyStore";
-import {ICompany} from "../../../actions/companyActionCreator";
-
-import CompanyStore from "../../../stores/companyStore";
-// I import SessionStore from "../../../stores/sessionStore";
+import companyActionCreator,
+    {ICompany} from "../../../actions/companyActionCreator";
 
 interface IShowCompaniesState {
     companies : Array<ICompany>;
```

Comment:
1 - Presumo proprio di si'
2 - Si, e' necessario fare il bind(this). Ho trovato scritto che:
"React components using ES6 classes no longer autobind this to non React methods."